lib/lightning/invocation.ex

```diff
@@ -347,54 +347,42 @@ defmodule Lightning.Invocation do
         end
       )
 
-    project
+    project_id
+    |> base_query()
     |> search_workorders_query(search_params)
     |> Repo.paginate(params)
   end
 
-  def count_workorders(%Project{} = project, search_params) do
-    project
-    |> count_workorders_query(search_params)
-    |> Repo.aggregate(:count)
-  end
-
-  def search_workorders_query(
+  def search_workorders_without_wiped_dataclips(
```

Comment:
Hmm, I'm not 100% getting why this gets a function to its own here.
I'm usually adverse to making function specifically for different cases like this, it can lead to divergence between the two differ variants.

I wasn't involved a whole bunch with the creation of this query stuff, and I don't actually have a strong enough stake in it (nor do I have a solid idea on what the alternative would be).

But the change in `run_live` [here](https://github.com/OpenFn/lightning/pull/2118/commits/5858e459b84d19b2639e96aa13471b124be99a98#diff-505fd6e66663185ea709cefb203e65dca8da3b7d0e1613b1dc1a778b82ca87ddL492) make me pause.

Happy to go with what Taylor thinks on this.

Reply:
Taking to another PR once there is a fix needed on existing query for workorders without wiped dataclips.